cellprofiler/modules/threshold.py

```diff
@@ -815,339 +817,89 @@ def run(self, workspace):
             statistics = workspace.display_data.statistics = []
             workspace.display_data.col_labels = ("Feature", "Value")
             if self.threshold_scope == TS_ADAPTIVE:
-                workspace.display_data.threshold_image = final_threshold
+                workspace.display_data.threshold_image = self.final_threshold
 
             for column in self.get_measurement_columns(workspace.pipeline):
                 value = workspace.measurements.get_current_image_measurement(column[1])
                 statistics += [(column[1].split("_")[1], str(value))]
 
-    def apply_threshold(self, image, threshold, automatic=False):
-        data = image.pixel_data
-
-        mask = image.mask
-
-        if not automatic and self.threshold_smoothing_scale.value == 0:
-            return (data >= threshold) & mask, 0
-
-        if automatic:
-            sigma = 1
-        else:
-            # Convert from a scale into a sigma. What I've done here
-            # is to structure the Gaussian so that 1/2 of the smoothed
-            # intensity is contributed from within the smoothing diameter
-            # and 1/2 is contributed from outside.
-            sigma = self.threshold_smoothing_scale.value / 0.6744 / 2.0
-
-        blurred_image = centrosome.smooth.smooth_with_function_and_mask(
-            data,
-            lambda x: scipy.ndimage.gaussian_filter(x, sigma, mode="constant", cval=0),
-            mask,
-        )
-
-        return (blurred_image >= threshold) & mask, sigma
-
-    def get_threshold(self, image, workspace, automatic=False):
-        need_transform = (
-                not automatic and
-                self.threshold_operation in (TM_LI, TM_OTSU) and
-                self.log_transform
-        )
-
-        if need_transform:
-            image_data, conversion_dict = centrosome.threshold.log_transform(image.pixel_data)
-        else:
-            image_data = image.pixel_data
+    def convert_setting(self, gui_setting_str):
```

Comment:
Not necessarily something to change now, but if we end up needing to do this a lot, should we make this a higher-level helper function?

Reply:
I agree. Would you envision a function such as this in `library_utils.py`?